policy-controller/src/main.rs

```diff
@@ -133,9 +139,58 @@ async fn main() -> Result<()> {
         probe_networks,
     });
 
+    // Fetch the policy-controller deployment so that we can use it as an owner
+    // reference of the Lease.
+    let api = k8s::Api::<Deployment>::namespaced(runtime.client(), &control_plane_namespace);
+    let deployment = api.get(&policy_deployment_name).await?;
+
+    let api = k8s::Api::namespaced(runtime.client(), &control_plane_namespace);
+    let params = PatchParams {
+        field_manager: Some("policy-controller".to_string()),
```

Comment:
not a blocker, just curious: i notice when we construct the kubert `LeaseManager`, we don't currently call [`with_field_manager`](https://docs.rs/kubert/latest/kubert/lease/struct.LeaseManager.html#method.with_field_manager): https://github.com/linkerd/linkerd2/blob/5f7bbbad32f4585996a79a9aca65dc8471d01897/policy-controller/src/main.rs#L139-L140

should we be calling that method as well?